AIO/src/org/aio/gui/Gui.java

```diff
@@ -263,7 +284,52 @@ private TaskPanel addTask(final TaskType taskType) {
                 gui.pack();
             });
         });
-        contextMenu.add(menuItem);
+
+        menuItemUp.addActionListener(e -> {
+            SwingUtilities.invokeLater(() -> {
+                int index = taskPanels.indexOf(taskPanelContent);
+                int replaceIndex = index - 1;
+                if(replaceIndex < 0){
+                    return;
+                }
+
+                Collections.swap(taskPanels,index, replaceIndex);
+
+                taskList.removeAll();
+                for(TaskPanelContent redrawTaskPanelContent : taskPanels){
+                    for(Component component : redrawTaskPanelContent.components){
+                        taskList.add(component);
+                    }
+                }
+
+                taskList.revalidate();
+                taskList.repaint();
+                gui.pack();
+            });
+        });
+
+        menuItemDown.addActionListener(e -> {
+            SwingUtilities.invokeLater(() -> {
```

Comment:
The move up/down is redundant and could  be consolidated into a function that takes a new index or offset.